flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -534,7 +536,70 @@ public int compare(JobStatusMessage o1, JobStatusMessage o2) {
 						"RunningJobs. Instead the response is of type " + result.getClass() + ".");
 			}
 		}
-		catch (Throwable t) {
+		catch (Exception t) {
+			return handleError(t);
+		}
+	}
+
+	/**
+	 * Executes the STOP action.
+	 * 
+	 * @param args Command line arguments for the stop action.
+	 */
+	protected int stop(String[] args) {
+		LOG.info("Running 'stop' command.");
+
+		StopOptions options;
+		try {
+			options = CliFrontendParser.parseStopCommand(args);
+		}
+		catch (CliArgsException e) {
+			return handleArgException(e);
+		}
+		catch (Exception t) {
+			return handleError(t);
+		}
+
+		// evaluate help flag
+		if (options.isPrintHelp()) {
+			CliFrontendParser.printHelpForStop();
+			return 0;
+		}
+
+		String[] stopArgs = options.getArgs();
+		JobID jobId;
+
+		if (stopArgs.length > 0) {
+			String jobIdString = stopArgs[0];
+			try {
+				jobId = new JobID(StringUtils.hexStringToByte(jobIdString));
+			}
+			catch (Exception e) {
+				LOG.error("Error: The value for the Job ID is not a valid ID.");
+				System.out.println("Error: The value for the Job ID is not a valid ID.");
+				return 1;
+			}
+		}
+		else {
+			LOG.error("Missing JobID in the command line arguments.");
+			System.out.println("Error: Specify a Job ID to stop a job.");
+			return 1;
+		}
+
+		try {
+			ActorGateway jobManager = getJobManagerGateway(options);
+			Future<Object> response = jobManager.ask(new StopJob(jobId), askTimeout);
+
+			Object rc = Await.result(response, askTimeout);
+
+			if (rc instanceof StoppingFailure) {
```

Comment:
I think that we decided not to use the built-in `Failure` object when transporting Exceptions between processes, if the Exceptions may be of user-defined classes (class loading issues). Using custom message types that explicitly require a `SerializedThrowable` makes it safer for users to recognize the need to supply a classloader.